codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -434,9 +453,10 @@ public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2E
         @Override
         public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promisedStreamId,
                 Http2Headers headers, int padding) throws Http2Exception {
-            Http2Stream parentStream = connection.requireStream(streamId);
+            Http2Stream parentStream = connection.stream(streamId);
 
-            if (streamCreatedAfterGoAwaySent(parentStream)) {
+            if (parentStream == null || streamCreatedAfterGoAwaySent(streamId)) {
```

Comment:
Good catch.  They don't actually say what the error is if the parent is missing, however. I'm guessing it's a connection error of type PROTOCOL_ERROR.